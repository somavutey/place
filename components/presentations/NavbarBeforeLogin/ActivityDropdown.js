import React, { useState } from "react";
import { serviceDropdown } from "./DropdownItem"; 
import Link from "next/link";
import { makeStyles } from "@mui/styles";

const useStyle = makeStyles({
  servicesSubmenu: {
    position: "absolute",
    top: "8vh",
    listStyle: "none",
    textAlign: "center",
    padding: 0,
    zIndex: '1',
  },

  li: {
    display: "block",
    width: "100px",
    height: "100%",
    textDecoration: "none",
    color: "white",

    background: "rgb(0, 212, 212)",
    cursor: "pointer",
    border: "1px solid",
    padding: "10px",
    "&:hover": {
      background: "rgb(0, 175, 175)",
    },
  },

  servicesSubmenuClicked: {
    display: "none",
  },
});

export default function ActivityDropdown() {
  const classes = useStyle();
  const [dropdown, setDropdown] = useState(false);
 
  return (
    <>
      <ul
        className={
          dropdown ? classes.servicesSubmenuClicked : classes.servicesSubmenu
        }
        onClick={() => setDropdown(!dropdown)}
      >
        {serviceDropdown.map((item) => {
          return (
            <li className={classes.li} key={item.id}>
              <Link
                href={item.path}
                // className={item.cName}
                onClick={() => setDropdown(false)}
              >
                {item.title}
              </Link>
            </li>
          );
        })}
      </ul>
    </>
  );
}
