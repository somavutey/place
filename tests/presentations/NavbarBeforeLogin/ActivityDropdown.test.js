import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ActivityDropdown from "./../../../components/presentations/NavbarBeforeLogin/ActivityDropdown";

it("ActivityDropdown", () => {
  const wrapper = shallow(<ActivityDropdown />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
