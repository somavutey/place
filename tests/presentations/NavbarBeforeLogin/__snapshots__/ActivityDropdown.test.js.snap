// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDropdown 1`] = `
<Fragment>
  <ul
    className="makeStyles-servicesSubmenu-1"
    onClick={[Function]}
  >
    <li
      className="makeStyles-li-2"
      key="1"
    >
      <Link
        href="./../../../FilterPagesBeforeLogin"
        onClick={[Function]}
      >
        Shopping
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="2"
    >
      <Link
        href="./../../../FilterPagesBeforeLogin"
        onClick={[Function]}
      >
        Swimming
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="3"
    >
      <Link
        href="./../../../FilterPagesBeforeLogin"
        onClick={[Function]}
      >
        Eating
      </Link>
    </li>
    <li
      className="makeStyles-li-2"
      key="4"
    >
      <Link
        href="./../../../FilterPagesBeforeLogin"
        onClick={[Function]}
      >
        Hiking
      </Link>
    </li>
  </ul>
</Fragment>
`;
